kdbg/debugger.cpp

```diff
@@ -2119,6 +2151,39 @@ void KDebugger::slotDisassemble(const QString& fileName, int lineNo)
     }
 }
 
+void KDebugger::setDisassemblyFlavor(QString flavor)
+{
+    // DefaultGenericFlavor -> "default"
+    if (flavor == DefaultGenericFlavor) {
+        flavor = m_globalFlavor;
+    }
+
+    if(m_tmpFlavor != flavor){
+	m_tmpFlavor = flavor;
+	m_d->executeCmd(DCsetdisassflavor, flavor);
+    }
+}
+
+void KDebugger::setDefaultFlavor(QString defFlavor)
+{
+    m_globalFlavor = defFlavor;
+
+    /*
+     * To specify a flavor for this program first read the global default.
+     * debugProgram will later set the (global) flavor. The user can later
+     * override the global flavor for this program. If the user reloaded a
+     * binary, `reloaded` is true and the current flavor is set forcefully
+     * to the global.
+     */
+    if (m_globalFlavor.isEmpty() && m_tmpFlavor.isEmpty()) {
+	m_tmpFlavor = DefaultX86Flavor;
+    } else if (m_tmpFlavor.isEmpty() ) {
+	m_tmpFlavor = m_globalFlavor;
+    }
+
+    setDisassemblyFlavor(m_tmpFlavor);
```

Comment:
Please let me suggest a simplification around `setDefaultFlavor`, `setDisassemblyFlavor`, and `m_tmpFlavor`.

The only point where the combination of the global settings and the program-specific settings are relevant, is when the command `DCsetdisassflavor` is submitted. That should be the only point where all the `if`s and `else`s should occur that compute which flavor the user wants.

Do not have a `m_tmpFlavor` that is already a combination of the two settings. Have only a `m_flavor` (in addition to `m_globalFlavor`) that reflects only the setting from *This Program*. In that dialog, the possible settings should be *intel*, *att*, *use global setting*, the latter being stored as `DefaultGenericFlavor`.

The function `KDebugger::setDisassemblyFlavor` could be replaced by `KDebugger::submitDisassemblyFlavor`, which does not have a parameter and computes the flavor to submit from `m_flavor` and `m_globalFlavor`.